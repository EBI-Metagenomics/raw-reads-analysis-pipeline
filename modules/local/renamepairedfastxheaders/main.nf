process RENAMEPAIREDFASTXHEADERS {
    tag "${meta.id}"
    label 'process_single'

    //conda "${moduleDir}/environment.yml"
    container "${workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container
        ? 'https://depot.galaxyproject.org/singularity/python:3.13'
        : 'quay.io/biocontainers/python:3.13'}"

    input:
    tuple val(meta), path(reads)
    path script

    output:
    tuple val(meta), path("*_renamed.*"), emit: reads
    path "versions.yml", emit: versions

    script:
    def args = task.ext.args ?: ''
    def (in_f1, in_f2) = reads

    def prefix1 = in_f1.getName().tokenize('.')[0]
    def ext1 = in_f1.getName().tokenize('.')[1..-1].join('.')
    if (ext1[-3..-1]=='.gz') {
        ext1 = ext1.tokenize('.')[0..-2].join('.')
    }
    def out_f1 = "${prefix1}_renamed.${ext1}"

    def prefix2 = in_f2.getName().tokenize('.')[0]
    def ext2 = in_f2.getName().tokenize('.')[1..-1].join('.')
    if (ext2[-3..-1]=='.gz') {
        ext2 = ext2.tokenize('.')[0..-2].join('.')
    }
    def out_f2 = "${prefix2}_renamed.${ext2}"
    """
    python ${script} ${args} --no-zip-output -f ${in_f1} -r ${in_f2} -o ${out_f1} -l ${out_f2}
    gzip ${out_f1} ${out_f2}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        renamepairedfastxheaders: \$(python --version |& sed '1!d ; s/python //')
    END_VERSIONS
    """

    stub:
    def (in_f1, in_f2) = reads

    def prefix1 = in_f1.tokenize('.')[0]
    def ext1 = in_f1.tokenize('.')[1..-1].join('.')
    def out_f1 = "${prefix1}_renamed${ext1}"

    def prefix2 = in_f2.tokenize('.')[0]
    def ext2 = in_f2.tokenize('.')[1..-1].join('.')
    def out_f2 = "${prefix2}_renamed${ext2}"

    """
    touch ${out_f1}
    touch ${out_f2}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        renamepairedfastxheaders: \$(python --version |& sed '1!d ; s/python //')
    END_VERSIONS
    """
}
