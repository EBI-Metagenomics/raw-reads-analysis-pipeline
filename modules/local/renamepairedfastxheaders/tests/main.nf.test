nextflow_process {

    name "Test Process RENAMEPAIREDFASTXHEADERS"
    script "../main.nf"
    process "RENAMEPAIREDFASTXHEADERS"

    tag "modules"
    tag "modules_"
    tag "renamepairedfastxheaders"

    test("mini sample should match snapshot") {

        when {
            process {
                """
                input[0] = [
		    ['id': 'test'],
		    [
		        file("${projectDir}/tests/fixtures/renamepairedfastxheaders/ERR10889147_mini_1.fq.gz", checkIfExists: true),
		        file("${projectDir}/tests/fixtures/renamepairedfastxheaders/ERR10889147_mini_2.fq.gz", checkIfExists: true),
		    ]
		]
		input[1] = file("${projectDir}/bin/rename_pe_reads.py")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("mini sample fasta should match snapshot") {

        when {
            process {
                """
                input[0] = [
		    ['id': 'test'],
		    [
		        file("${projectDir}/tests/fixtures/renamepairedfastxheaders/ERR10889147_mini_1.fa.gz", checkIfExists: true),
		        file("${projectDir}/tests/fixtures/renamepairedfastxheaders/ERR10889147_mini_2.fa.gz", checkIfExists: true),
		    ]
		]
		input[1] = file("${projectDir}/bin/rename_pe_reads.py")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("unbalanced (1>2) synthetic sample should match snapshot") {

        when {
            process {
                """
                input[0] = [
		    ['id': 'test'],
		    [
		        file("${projectDir}/tests/fixtures/renamepairedfastxheaders/synthetic_test_ub1_1.fq.gz", checkIfExists: true),
		        file("${projectDir}/tests/fixtures/renamepairedfastxheaders/synthetic_test_ub1_2.fq.gz", checkIfExists: true),
		    ]
		]
		input[1] = file("${projectDir}/bin/rename_pe_reads.py")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("unbalanced (2>1) synthetic sample should match snapshot") {

        when {
            process {
                """
                input[0] = [
		    ['id': 'test'],
		    [
		        file("${projectDir}/tests/fixtures/renamepairedfastxheaders/synthetic_test_ub2_1.fq.gz", checkIfExists: true),
		        file("${projectDir}/tests/fixtures/renamepairedfastxheaders/synthetic_test_ub2_2.fq.gz", checkIfExists: true),
		    ]
		]
		input[1] = file("${projectDir}/bin/rename_pe_reads.py")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("mismatching names sythethic sample should match snapshot") {

        when {
            process {
                """
                input[0] = [
		    ['id': 'test'],
		    [
		        file("${projectDir}/tests/fixtures/renamepairedfastxheaders/synthetic_test_mm_1.fq.gz", checkIfExists: true),
		        file("${projectDir}/tests/fixtures/renamepairedfastxheaders/synthetic_test_mm_2.fq.gz", checkIfExists: true),
		    ]
		]
		input[1] = file("${projectDir}/bin/rename_pe_reads.py")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
