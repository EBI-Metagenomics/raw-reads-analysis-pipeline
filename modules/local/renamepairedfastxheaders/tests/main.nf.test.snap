{
    "unbalanced (2>1) synthetic sample should match snapshot": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "synthetic_test_ub2_1_renamed.fq.gz:md5,1df5a8a41907b5e09ddf9d075b965a0f",
                            "synthetic_test_ub2_2_renamed.fq.gz:md5,1f41a3256abd4e5571bf8249342a1167"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,ee72c9395597d6890ccc37b29af648d2"
                ],
                "reads": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "synthetic_test_ub2_1_renamed.fq.gz:md5,1df5a8a41907b5e09ddf9d075b965a0f",
                            "synthetic_test_ub2_2_renamed.fq.gz:md5,1f41a3256abd4e5571bf8249342a1167"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,ee72c9395597d6890ccc37b29af648d2"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2025-05-20T10:46:07.622798241"
    },
    "mini sample should match snapshot": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "ERR10889147_mini_1_renamed.fq.gz:md5,428efc43309474dad52a9bef226746a3",
                            "ERR10889147_mini_2_renamed.fq.gz:md5,78a51b1e1e7fc754cf95406e226dec87"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,ee72c9395597d6890ccc37b29af648d2"
                ],
                "reads": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "ERR10889147_mini_1_renamed.fq.gz:md5,428efc43309474dad52a9bef226746a3",
                            "ERR10889147_mini_2_renamed.fq.gz:md5,78a51b1e1e7fc754cf95406e226dec87"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,ee72c9395597d6890ccc37b29af648d2"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2025-05-20T10:21:18.735445608"
    },
    "unbalanced (1>2) synthetic sample should match snapshot": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "synthetic_test_ub1_1_renamed.fq.gz:md5,db846a2b67a2a74a4c029f88c5d579ac",
                            "synthetic_test_ub1_2_renamed.fq.gz:md5,5032b40f81fd292faa7890c12778896f"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,ee72c9395597d6890ccc37b29af648d2"
                ],
                "reads": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "synthetic_test_ub1_1_renamed.fq.gz:md5,db846a2b67a2a74a4c029f88c5d579ac",
                            "synthetic_test_ub1_2_renamed.fq.gz:md5,5032b40f81fd292faa7890c12778896f"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,ee72c9395597d6890ccc37b29af648d2"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2025-05-20T10:45:40.388625138"
    },
    "mismatching names sythethic sample should match snapshot": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "synthetic_test_mm_1_renamed.fq.gz:md5,428efc43309474dad52a9bef226746a3",
                            "synthetic_test_mm_2_renamed.fq.gz:md5,fb308932d4346e4f607f553c6fafa732"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,ee72c9395597d6890ccc37b29af648d2"
                ],
                "reads": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "synthetic_test_mm_1_renamed.fq.gz:md5,428efc43309474dad52a9bef226746a3",
                            "synthetic_test_mm_2_renamed.fq.gz:md5,fb308932d4346e4f607f553c6fafa732"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,ee72c9395597d6890ccc37b29af648d2"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2025-05-20T10:46:41.171993024"
    },
    "mini sample fasta should match snapshot": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "ERR10889147_mini_1_renamed.fa.gz:md5,7086c051b71f335e9df64a33d6aad697",
                            "ERR10889147_mini_2_renamed.fa.gz:md5,492a1401a7379cc63b4133dcef60c971"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,ee72c9395597d6890ccc37b29af648d2"
                ],
                "reads": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "ERR10889147_mini_1_renamed.fa.gz:md5,7086c051b71f335e9df64a33d6aad697",
                            "ERR10889147_mini_2_renamed.fa.gz:md5,492a1401a7379cc63b4133dcef60c971"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,ee72c9395597d6890ccc37b29af648d2"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2025-05-20T10:45:02.50036334"
    }
}